### ISSUE TITLE
Example for Typescript and SubCommands

### ISSUE THREAD

**Issue by dibog** [yargs/yargs#1940]
Are there any examples how to work with sub commands and typescript?

**Comment by bcoe** [comment]
@dibog I do not believe there are currently any examples. My expectation would be that it works similarly to plain JavaScript, but I'm not sure how smart it is about propagating typing.

**Comment by justinmchase** [comment]
Is there a plain javascript example of sub commands?

**Comment by aspiers** [comment]
This is a big hole in the docs :-(  TypeScript is hugely prevalent these days, yet apparently the only TypeScript docs are https://github.com/yargs/yargs/blob/main/docs/typescript.md which are very short and don't provide any explanation of how to use things like the `CommandModule` type.

See also https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57685
